pubpub/platform
# feat: add title to pubs table on pub_value update

- **feat: add title field to pubs table and update updatedAt trigger**
- **feat: make the title updates work**

## Issue(s) Resolved
#736 

## High-level Explanation of PR

This PR adds a new `title` column to the `pubs` table. This column is not to be updated manually, but to be updated through newly added triggers on inserts/updates/deletes to the `pub_value` table. A migration is added to backfill these title values to existing pubs.

Additionally, it modifies the `getPubTitle` function and places where it's called to use this new field.
The fallback for searching for fields with `title` in their slug is kept in place.

### Possible improvements

I could make it so that instead of having `title: null` for no title, I could set it to be `Untitled ${PubType} - ${createdAt}` instead. This would make searching easier, but would make it harder to distinguish pubs without a title from those with one in code (ie instead of doing `title === null`, you'd have to do `!title.match(/Untitled/)`, which is much more fragile).

## Test Plan
1. Run tests
2. Create a new field, eg `arcadia-research:name`
3. Create a new pubtype, eg `Person`, with `arcadia-research:name` as its title
4. Create a new `Person` pub, filling out the `name` field.
5. See that on `/stages`, `/stages/manage`, `/pubs`, and `/pubs/<pubId>` page the title is set to be that new value.
6. Create another `Person` pub, this time without filling in the title.
7. See that the title is shown as `Untitled Person - ${createdAt}`, rather than `Untitled Pub`.



## Screenshots (if applicable)

## Notes

The types for `PubTitle` props etc are a bit ugly. I address this in #823
